scripts/zones/Newton_Movalpolos/npcs/qm_collector.lua

```diff
@@ -0,0 +1,17 @@
+-----------------------------------
+-- Area: Newton_Movalpolos
+--  NPC: ??? for Goblin Collector
+-----------------------------------
+local ID = require("scripts/zones/Newton_Movalpolos/IDs")
+require("scripts/globals/npc_util")
+-----------------------------------
+
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    if npcUtil.tradeHas(trade, xi.items.PREMIUM_BAG) and npcUtil.popFromQM(player, npc, ID.mob.GOBLIN_COLLECTOR) then
```

Comment:
use `npcUtil.tradeHasExactly` unless the trade actually works with other items, so long as the premium bad is in it.
If it does, please add a comment so we know it's supposed to do that.